examples/instruct_pix2pix/train_instruct_pix2pix_xl.py

```diff
@@ -0,0 +1,1113 @@
+                # check out the section 3.2.1 of the original paper https://arxiv.org/abs/2211.09800.
+                if args.conditioning_dropout_prob is not None:
+                    random_p = torch.rand(bsz, device=latents.device, generator=generator)
+                    # Sample masks for the edit prompts.
+                    prompt_mask = random_p < 2 * args.conditioning_dropout_prob
+                    prompt_mask = prompt_mask.reshape(bsz, 1, 1)
+                    # Final text conditioning.
+                    ### Begin: Get null conditioning
+                    null_conditioning_list = []
+                    for a_tokenizer, a_text_encoder in zip(
+                        (tokenizer_1, tokenizer_2), (text_encoder_1, text_encoder_2)
+                    ):
+                        null_conditioning_list.append(
+                            a_text_encoder(
+                                tokenize_captions([""], a_tokenizer=a_tokenizer).to(accelerator.device),
+                                output_hidden_states=True,
+                            ).hidden_states[-2]
+                        )
+                    ### End: Get null conditioning
+                    null_conditioning = torch.concat(null_conditioning_list, dim=-1)
+                    encoder_hidden_states = torch.where(prompt_mask, null_conditioning, encoder_hidden_states)
+
+                    # Sample masks for the original images.
+                    image_mask_dtype = original_image_embeds.dtype
+                    image_mask = 1 - (
+                        (random_p >= args.conditioning_dropout_prob).to(image_mask_dtype)
+                        * (random_p < 3 * args.conditioning_dropout_prob).to(image_mask_dtype)
+                    )
+                    image_mask = image_mask.reshape(bsz, 1, 1, 1)
+                    # Final image conditioning.
+                    original_image_embeds = image_mask * original_image_embeds
+
+                # Concatenate the `original_image_embeds` with the `noisy_latents`.
+                concatenated_noisy_latents = torch.cat([noisy_latents, original_image_embeds], dim=1)
+
+                # Get the target for loss depending on the prediction type
+                if noise_scheduler.config.prediction_type == "epsilon":
+                    target = noise
+                elif noise_scheduler.config.prediction_type == "v_prediction":
+                    target = noise_scheduler.get_velocity(latents, noise, timesteps)
+                else:
+                    raise ValueError(f"Unknown prediction type {noise_scheduler.config.prediction_type}")
+
+                ### Begin SDXL
+                add_text_embeds = pooled_prompt_embeds
+
+                crops_coords_top_left = (0, 0)
+                target_size = (args.resolution, args.resolution)
+                original_size = original_image_embeds.shape[-2:]
```

Comment:
Why are we determining the original image resolution from the image embeddings?

Reply:
This might be a bug. Fixed already.